pandas/tests/indexes/multi/test_setops.py

```diff
@@ -578,6 +578,30 @@ def test_union_keep_ea_dtype(any_numeric_ea_dtype, val):
     tm.assert_index_equal(result, expected)
 
 
+def test_union_with_duplicates_keep_ea_dtype(any_numeric_ea_dtype):
+    # GH48900
+    mi1 = MultiIndex.from_arrays(
```

Comment:
What happens if theres duplicate `NA`s?

Reply:
Works the same as non-NA. I added the dupe NA case to the test.